pkg/kv/mem/store.go

```diff
@@ -92,40 +118,49 @@ func (s *Store) SetIf(_ context.Context, key, value []byte, valuePredicate kv.Pr
 	}
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	curr, currOK := s.m[string(key)]
+	combinedKey := combinedKey(partitionKey, key)
+
+	curr, currOK := s.m[combinedKey]
 	if valuePredicate == nil {
 		if currOK {
 			return fmt.Errorf("key=%v: %w", key, kv.ErrPredicateFailed)
 		}
-		s.insertNewKey(key)
+		s.insertNewKey(combinedKey)
 	} else if !bytes.Equal(valuePredicate.([]byte), curr) {
-		return fmt.Errorf("%w: key=%v", kv.ErrPredicateFailed, key)
+		return fmt.Errorf("%w: key=%v", kv.ErrPredicateFailed, combinedKey)
 	}
-	s.m[string(key)] = value
+	s.m[combinedKey] = value
 	return nil
 }
 
-func (s *Store) Delete(_ context.Context, key []byte) error {
+func (s *Store) Delete(_ context.Context, partitionKey, key []byte) error {
+	if len(partitionKey) == 0 {
+		return kv.ErrMissingPartitionKey
+	}
 	if len(key) == 0 {
 		return kv.ErrMissingKey
 	}
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	if _, found := s.m[string(key)]; !found {
+	combinedKey := combinedKey(partitionKey, key)
+
+	if _, found := s.m[combinedKey]; !found {
 		return nil
 	}
-	idx := sort.SearchStrings(s.keys, string(key))
-	if idx < len(s.keys) && s.keys[idx] == string(key) {
+	idx := sort.SearchStrings(s.keys, combinedKey)
+	if idx < len(s.keys) && s.keys[idx] == combinedKey {
 		s.keys = append(s.keys[:idx], s.keys[idx+1:]...)
 	}
-	delete(s.m, string(key))
+	delete(s.m, combinedKey)
 	return nil
 }
 
-func (s *Store) Scan(_ context.Context, start []byte) (kv.EntriesIterator, error) {
+func (s *Store) Scan(_ context.Context, partitionKey, start []byte) (kv.EntriesIterator, error) {
+	combinedKey := combinedKey(partitionKey, start)
```

Comment:
I think that in case of `partitionKey = ""` we will scan all keys with no partition key, or all keys with partition key starting with lower case. All partition keys starting with upper case will be skipped as they preced `_`